### ISSUE TITLE
Terminating/Completed pods should be filtered from the CLI stat output

### ISSUE THREAD

**Issue by klingerf** [runconduit/conduit#1010]
**NOTE**: this relates to #1009. It's possible the right fix for both issues to do the filtering in the public-api prior to sending back the list of pods, so that we don't have to implement filtering in both the web frontend and the CLI.

---

The conduit stat output for shows pods that are terminating or completed (e.g. after a successful job run). I think we should filter them out of the CLI output. Completed pods aren't displayed by default by kubectl, and terminating pods introduce confusion because there's no visual indication that they're in the process of terminating.

Here's an example of a terminating pod:

```
$ conduit -n conduit stat po
NAME                          MESHED   SUCCESS      RPS   LATENCY_P50   LATENCY_P95   LATENCY_P99   SECURED
controller-755995f5d5-5gzn8      1/1   100.00%   0.8rps           8ms         375ms         395ms        0%
controller-755995f5d5-gm5c4      1/1   100.00%   0.7rps          20ms         260ms         292ms        0%
controller-755995f5d5-wgbhq      0/0         -        -             -             -             -         -
controller-755995f5d5-wj6z9      1/1   100.00%   0.9rps          58ms         178ms         196ms        0%
grafana-85cd9c64d-5hpjx          1/1   100.00%   0.1rps           5ms          10ms          10ms        0%
prometheus-7ff65b5cbd-8vqzl      1/1   100.00%   8.6rps          28ms         153ms         271ms        0%
web-84df49b76b-wgfd2             1/1   100.00%   1.4rps         145ms         262ms         292ms        0%
```

Interestingly, the "meshed" pod count for the conduit deployment doesn't count that pod among it's total number of pods for the controller:

```
$ conduit -n conduit stat deploy
NAME         MESHED   SUCCESS      RPS   LATENCY_P50   LATENCY_P95   LATENCY_P99   SECURED
controller      3/3   100.00%   2.4rps          94ms         395ms         479ms        0%
grafana         1/1   100.00%   0.1rps           5ms          10ms          10ms        0%
prometheus      1/1   100.00%   8.2rps          48ms         280ms         377ms        0%
web             1/1   100.00%   1.4rps         192ms         875ms         975ms        0%
```

And here's an example of a completed pod in the stat output:

```
$ conduit stat pods
NAME                     MESHED   SUCCESS   RPS   LATENCY_P50   LATENCY_P95   LATENCY_P99   SECURED
mysql-79984d68c4-fdl6x      0/1         -     -             -             -             -         -
mysql-init-7zgqr            0/0         -     -             -             -             -         -
```

```
$ kubectl get po --show-all
NAME                     READY     STATUS      RESTARTS   AGE
mysql-79984d68c4-fdl6x   1/1       Running     0          1m
mysql-init-7zgqr         0/1       Completed   0          1m
```

**Comment by ihcsim** [comment]
@klingerf I am hoping to help out with this issue. My thought is to do the filtering in the `grpcServer.resourceQuery()` method as seen in [my branch](https://github.com/ihcsim/conduit/commit/28653921d8941d70822216513df8fac725401ab3). 

I feel any other changes in the public-api will require changing [`Lister.GetObjects()`](https://github.com/runconduit/conduit/blob/ec2433e9bdde6d3adc8efb21c058c75d4ca512c5/controller/k8s/lister.go#L101) to accept some filters. I am reluctant with this approach because of the generic nature of this method.

Tap seems to do it differently by using the [`Lister.GetPodsFor()`](https://github.com/runconduit/conduit/blob/ec2433e9bdde6d3adc8efb21c058c75d4ca512c5/controller/tap/server.go#L49-L62) method. But I don't think this method is suitable here.

If my approach is acceptable, I will submit a PR. WDYT?

**Comment by klingerf** [comment]
@ihcsim Awesome! Thanks for helping out with this issue. I think you're definitely on the right track. And hooray for tests!

My preference is to make this change in the `getPods` method itself, in lister.go. As far as I can tell, we used to have a check inside that method, but then we dropped it when the method was moved to lister.go, here:

https://github.com/runconduit/conduit/pull/835/files#diff-094b2b3ba038efd849e3cfa81be8a0fcL426

Based on my reading of the code, if we modify `getPods` to re-introduce the pending-or-running check, then that means:

* Completed/Terminating pods will no longer show up in `conduit stat` and the web UI
* Completed/Terminating pods will no longer be tap-able

From my perspective, this seems like the right behavior. I don't think we need to introduce a param to conditionally filter these pods; we should just filter them in all cases, similar to what the `GetPodsFor` in lister.go is already doing.

I'd be more than happy to review your PR whenever it's ready. Thanks again for working on this.
